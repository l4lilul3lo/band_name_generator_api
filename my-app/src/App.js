import { useState } from "react";
import "./App.css";
import axios from "axios";
import BandName from "./component/BandName";
import BandNameList from "./component/BandNameList";
import { useEffect } from "react/cjs/react.development";
const App = () => {
  const [bandname, setBandname] = useState();
  const [triggerApiCall, setTriggerApiCall] = useState(false);

  function handleClick(e) {
    e.preventDefault();
    setTriggerApiCall(true);
    setBandname(undefined);
  }

  useEffect(() => {
    if (!bandname && triggerApiCall === true) {
      setTriggerApiCall(false);
      axios.get("http://localhost:8000/bandname/random").then((resp) => {
        setBandname(resp.data.bandName);
      });
    }
  }, [bandname, triggerApiCall]);

  return (
    <div className="App">
      <header className="App-header">
        <button
          className="App-button"
          type="button"
          id="button"
          onClick={handleClick}
        >
          What should we call the band?
        </button>
        <span className="App-message">
          {bandname != null ? (
            <BandName bandname={bandname} setBandname={setBandname} />
          ) : null}
        </span>
        <BandNameList />
      </header>
    </div>
  );
};

export default App;
