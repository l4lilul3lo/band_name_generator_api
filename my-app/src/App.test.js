import { fireEvent, render, waitFor } from "@testing-library/react";
import App from "./App";

describe("App component unit tests", () => {
  it("Smoke Test", () => {
    render(<App />);
  });
  it("Snapshot tests", () => {
    const screen = render(<App />);
    expect(screen.asFragment()).toMatchSnapshot();
  });
  it("should generate random bandname when button is clicked", async () => {
    const screen = render(<App />);
    const generateBtn = screen.getByTestId("generate-btn");
    fireEvent.click(generateBtn);

    await waitFor(() => {
      screen.debug();
      const addBtn = screen.getByTestId("add-band-btn");
      screen.debug();
      expect(addBtn).toBeInTheDocument();
    });
  });
  it.todo(
    "a list of bandnames should appear when list-bandnames-btn is clicked"
  );
});
