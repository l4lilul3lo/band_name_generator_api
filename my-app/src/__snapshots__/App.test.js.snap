// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component unit tests Snapshot tests 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <button
        class="App-button"
        data-testid="generate-btn"
        id="button"
        type="button"
      >
        What should we call the band?
      </button>
      <span
        class="App-message"
        data-testid="band-name"
      />
      <button
        class="App-button"
        id="button"
        type="button"
      >
         See a list of upvoted band names!
      </button>
      <ul />
    </header>
  </div>
</DocumentFragment>
`;
